source/common/router/header_formatter.h

```diff
@@ -22,10 +22,13 @@ class HeaderFormatter {
   virtual const std::string format(const Envoy::StreamInfo::StreamInfo& stream_info) const PURE;
 
   /**
-   * @return bool indicating whether the formatted header should be appended to the existing
-   *              headers or replace any existing values for the header
+   * @return envoy::config::core::v3::HeaderValueOption::HeaderAppendAction describes what append
+   * action to take indicating whether:
+   * 1. The formatted header should be added only if it's not already present;
+   * 2. New value should be appended to the existing values if header already exists; or
+   * 3. New value should be overwritten by discarding any existing values if header already exists;
```

Comment:
I don't think you need to re-document this. The enum is self explanatory.

Reply:
Removed.